content/sensu-core/2.0/reference/hooks.md

```diff
@@ -0,0 +1,96 @@
+---
+title: "Hooks"
+description: "The hooks reference guide."
+weight: 1
+version: "2.0"
+product: "Sensu Core"
+platformContent: true
+menu:
+  sensu-core-2.0:
+    parent: reference
+---
+
+# Hooks
+
+### What are hooks?
+
+Hooks are commands run by the Sensu client in response to the result of a check, mutator, or handler execution. The Sensu client will execute the appropriate configured hook command, depending on the execution status (e.g. 1). The hook command output, status, executed timestamp, and duration are captured and published in the event result. Hook commands can optionally receive JSON serialized Sensu client data via STDIN.
+
+### New and improved hooks!
+
+In 2.x, we've redesigned and expanded on the concept of 1.x check hooks. Hooks are now their own resource, and can be created and managed independent of the check configuration scope. With unique and descriptive identifiers, hooks are now reusable and can respond to checks, mutators and handlers! And thats not all, you can now execute multiple hooks for any given response code.
+
+Check out Sean's [blog post](https://blog.sensuapp.org/using-check-hooks-a739a362961f) about Sensu Core check hooks to see how you can use Sensu for auto-remediation tasks!
+
+### Viewing
+
+To view all the hooks that are currently configured for the cluster, enter:
+
+{{< highlight shell >}}
+sensuctl hook list
+{{< /highlight >}}
+
+If you want more details on a hook, the `info` subcommand can help you out.
+
+> sensuctl hook info nginx-start
+{{< highlight shell >}}
+=== nginx-start
+Name:           nginx-start
+Command:        sudo /etc/init.d/nginx start
+Timeout:        60
+Stdin?:         false
+Organization:   default
+Environment:    default
+{{< /highlight >}}
+
+### Management
+
+Hooks can be configured both interactively or by using CLI flags.
+
+{{< highlight shell >}}
+sensuctl hook create nginx-start --command "sudo /etc/init.d/nginx start" --timeout 10
+{{< /highlight >}}
+
+To delete an existing hook, simply pass the name of the hook to the `delete` command.
+
+{{< highlight shell >}}
+sensuctl hook delete nginx-start
+{{< /highlight >}}
+
+## Check hooks
+
+### What is a check hook?
+
+A check hook is a type of hook that lives in the check configuration scope. Check hooks associate existing hooks to a check and type. The check hook type is defined by the response code or severity of the check execution result. Valid check hook types include (in order of precedence): “1”-“255”, “ok”, “warning”, “critical”, “unknown”, and “non-zero”.
```

Comment:
If "ok" is a valid hook type, then why isn't 0-255 allowed instead of 1-255?

Reply:
You bring up a good point! It's simply how 1.x is implemented, so I'm not sure why "ok" is the only way to represent a "0" response code.

Reply:
@grepory or @portertech can you comment on this?

Reply:
@portertech can we revisit this interface?

Reply:
The 1.x hook logic and config validator does allow 0-255, this is probably a doc oversight:

Hook code: https://github.com/sensu/sensu/blob/e895d5958f1a3857601d3ef321f59dfc4d9e8f1f/lib/sensu/client/process.rb#L133-L134

Validator code: https://github.com/sensu/sensu-settings/blob/aa818a30ae7ec9722929a3c5c6667f236b0ad05b/lib/sensu/settings/validators/check.rb#L201

Reply:
Oh funny. I’ll write up a ticket to make that quick change tomorrow!

Reply:
It was actually a doc oversight in 2.x too 🤣 Our regex includes 0, but I'm pushing a change to update our validation message with a minor bug fix!

```
// CheckHookRegexStr used to validate type of check hook
var CheckHookRegexStr = `([0-9]|[1-9][0-9]|1[0-9][0-9]|2[0-4][0-9]|25[0-5])`
```